src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -1195,86 +1237,111 @@ int fx_muxer_t::read_config_and_execute(
         trace::error(_X("The specified deps.json [%s] does not exist"), deps_file.c_str());
         return StatusCode::InvalidArgFailure;
     }
-    if (!runtime_config.empty() && (!pal::realpath(&runtime_config) || !pal::file_exists(runtime_config)))
-    {
-        trace::error(_X("The specified runtimeconfig.json [%s] does not exist"), runtime_config.c_str());
-        return StatusCode::InvalidConfigFile;
-    }
 
-    pal::string_t config_file, dev_config_file;
+    // Read config
+    // The fx_definitions_owned variable controls object lifetime
+    std::vector<std::unique_ptr<fx_definition_t>> fx_definitions_owned;
+    std::vector<fx_definition_t*> fx_definitions;
 
-    if (runtime_config.empty())
-    {
-        trace::verbose(_X("App runtimeconfig.json from [%s]"), app_candidate.c_str());
-        get_runtime_config_paths_from_app(app_candidate, &config_file, &dev_config_file);
-    }
-    else
-    {
-        trace::verbose(_X("Specified runtimeconfig.json from [%s]"), runtime_config.c_str());
-        get_runtime_config_paths_from_arg(runtime_config, &config_file, &dev_config_file);
-    }
+    auto app = new fx_definition_t();
+    fx_definitions_owned.push_back(std::unique_ptr<fx_definition_t>(app));
+    fx_definitions.push_back(app);
 
-    runtime_config_t config(config_file, dev_config_file);
-    if (!config.is_valid())
+    int rc = read_config(*app, app_candidate, runtime_config);
+    if (rc)
     {
-        trace::error(_X("Invalid runtimeconfig.json [%s] [%s]"), config.get_path().c_str(), config.get_dev_path().c_str());
-        return StatusCode::InvalidConfigFile;
+        return rc;
     }
 
-    // Append specified probe paths first and then config file probe paths into realpaths.
-    std::vector<pal::string_t> probe_realpaths;
-    for (const auto& path : spec_probe_paths)
-    {
-        append_probe_realpath(path, &probe_realpaths, config.get_tfm());
-    }
-    for (const auto& path : config.get_probe_paths())
-    {
-        append_probe_realpath(path, &probe_realpaths, config.get_tfm());
-    }
+    auto config = app->get_runtime_config();
 
     // 'Roll forward on no candidate fx' is disabled by default. It can be enabled through:
     // 1. Command line argument (--roll-forward-on-no-candidate-fx)
     // 2. Runtimeconfig json file ('rollForwardOnNoCandidateFx' property)
     // 3. DOTNET_ROLL_FORWARD_ON_NO_CANDIDATE_FX env var
     // The conflicts will be resolved by following the priority rank described above (from 1 to 3).
     // The env var condition is verified in the config file processing
-    int roll_fwd_on_no_candidate_fx_val = config.get_roll_fwd_on_no_candidate_fx();
     if (!roll_fwd_on_no_candidate_fx.empty())
     {
-        roll_fwd_on_no_candidate_fx_val = pal::xtoi(roll_fwd_on_no_candidate_fx.c_str());
+        config.set_get_roll_fwd_on_no_candidate_fx(pal::xtoi(roll_fwd_on_no_candidate_fx.c_str()));
```

Comment:
(nit) `set_get_` is kind of a strange name.  Why not just `set_`?

Reply:
Fixed (copy-paste issue) Thanks